File: 3782.last-remaining-integer-after-alternating-deletion-operations.py
#
# @lc app=leetcode id=3782 lang=python3
#
# [3782] Last Remaining Integer After Alternating Deletion Operations
#
# @lc code=start
class Solution:
    def lastInteger(self, n: int) -> int:
        return self.josephus(n)
    
    def josephus(self, n: int) -> int:
        if n == 1:
            return 1
        else:
            return (self.josephus(n - 1) + 2) % n or n
# @lc code=end

File: 3782.last-remaining-integer-after-alternating-deletion-operations.cpp
# @lc app=leetcode id=3782 lang=cpp
#
# [3782] Last Remaining Integer After Alternating Deletion Operations
#
# @lc code=start
class Solution {
public:
    long long lastInteger(long long n) {
        if (n == 1) return 1;
        return 2 * lastInteger(n / 2) - (n % 2 == 0 ? 1 : -1);
    }
};
# @lc code=end